RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFMultiDepthClusterWarpIntrinsics.h

```diff
@@ -0,0 +1,219 @@
+#ifndef PFClusterProducer_plugins_alpaka_PFMultiDepthClusterWarpIntrinsics_h
+#define PFClusterProducer_plugins_alpaka_PFMultiDepthClusterWarpIntrinsics_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+#include <type_traits>
+
+namespace cms::alpakatools {
+  namespace warp {
+
+#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+    using warp_mask_t = unsigned;  // 32-bit masks on NVIDIA
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    using warp_mask_t = unsigned long;  // 64-bit masks on AMD (check!)
+#endif
+
+#else
+    using warp_mask_t = unsigned;  // for host (no-op)
+#endif
+
+    template <typename TAcc, typename = std::enable_if_t<alpaka::isAccelerator<TAcc>>>
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE void syncWarpThreads_mask(TAcc const& acc, warp::warp_mask_t mask) {
+      if (mask == 0)
+        return;  //early return for the trivial mask
+
+#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+      // Alpaka CUDA backend
+      __syncwarp(mask);  // Synchronize all threads within a subset of lanes in the warp
+#endif
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+      // Alpaka HIP backend
+      __builtin_amdgcn_wave_barrier();
```

Comment:
Now that we have ROCm 7.0 in CMSSW, `__syncwarp()` and `__syncwarp(mask)` are available also for HIP/ROCm.